### PR TITLE
🌱 test: add coverage to find issues running on the next kubernetes release as management cluster

### PR DESCRIPTION
**Pull request by chrischdi**
…ase as management cluster

**What this PR does / why we need it**:

This helps to find upcoming issues like the one fixed in:

- #10972

way more earlier in the cycle and not when we start bumping to beta or rc's for the management cluster.

/area e2e-testing

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by chrischdi** [comment]
/retitle 🌱 [wip] test: add coverage to find issues running on the next kubernetes release as management cluster

We have to build the kind image.

File: test/e2e/clusterctl_upgrade_test.go [hunk 1]
```diff
@@ -325,3 +326,31 @@ var _ = Describe("When testing clusterctl upgrades using ClusterClass (v1.7=>cur
 		}
 	})
 })
+
+var _ = Describe("When testing clusterctl upgrades using ClusterClass (v1.7=>current) on latest management cluster [ClusterClass]", func() {
```
Comment (chrischdi) [inline test/e2e/clusterctl_upgrade_test.go:330]: This test would now run on e2e-mink8s and on full e2e.

Maybe we want to add some GINKGO_SKIP/GINKGO_FOCUS to only run it on full e2e.

cc @sbueringer

**Comment by chrischdi** [comment]
/hold

For discussion on https://github.com/kubernetes-sigs/cluster-api/pull/10984#discussion_r1701769686

File: test/e2e/clusterctl_upgrade_test.go (hunk 1 above)
Comment (sbueringer) [inline test/e2e/clusterctl_upgrade_test.go:330]:
```suggestion
var _ = Describe("When testing clusterctl upgrades using ClusterClass (v1.7=>current) on K8S latest ci mgmt cluster [ClusterClass]", func() {
```
Just for consistency with the other test using that version

Comment (sbueringer) [reply test/e2e/clusterctl_upgrade_test.go:330]: Given that it should just work I would keep it simple and just run it in both jobs

**Comment by sbueringer** [comment]
/lgtm
/approve

Feel free to merge

**Comment by sbueringer** [comment]
/hold

Maybe we're now really hitting the 2h timeouts regularly :)

Let's get more data and then talk about in on Monday (maybe we should increase the timeout & interval to 3h)

**Comment by sbueringer** [comment]
We've hit the timeout again

/test pull-cluster-api-e2e-main
/test pull-cluster-api-e2e-mink8s-main

**Comment by sbueringer** [comment]
Talked to Christian, we're going to increase the timeouts & intervals to 3h

**Comment by chrischdi** [comment]
xref: https://github.com/kubernetes/test-infra/pull/33220

**Comment by sbueringer** [comment]
Merged

/hold cancel

/override pull-cluster-api-e2e-mink8s-main 
(just a timeout, also worked before)

**Comment by sbueringer** [comment]
/hold cancel
(the job should now block merge)

**Pull request merged**